frontend/catalyst/qfunc.py

```diff
@@ -89,6 +89,42 @@ def extract_backend_info(
     def __call__(self, *args, **kwargs):
         assert isinstance(self, qml.QNode)
 
+        # Mid-circuit measurement configuration/execution
+        mcm_config = self.execute_kwargs["mcm_config"]
+        _dynamic_one_shot_called = kwargs.pop("_dynamic_one_shot_called", False)
```

Comment:
I don't know that I like this being passes as a call argument to the qnode. I would prefer if it was a property in that case. Or maybe it's a special shot value? Or a subclass of QNode?

Reply:
I've made them temporary attributes of the qnode that get deleted at the end of the execution. Let me know what you think about that implementation.